valkyrie/lib/valkyrie/storage/fedora.rb

```diff
@@ -46,6 +46,21 @@ def delete(id:)
       ActiveFedora::File.new(active_fedora_identifier(id: id)).ldp_source.delete
     end
 
+    def supports_versions?
+      true
+    end
+
+    def versions(resource:)
+      ActiveFedora::File.new(active_fedora_identifier(id: resource.id)).versions.all.map(&:label)
+    end
+
+    def retrieve_version(resource:, label:)
```

Comment:
This dependence on `label:` which I THINK is auto-generated makes me a little uncomfortable. Is this a string always? Can we assign versions identifiers? Plus, Fedora 4's moving to timestamps instead of labels, so if that's where the label implementation comes from we probably shouldn't use it.

Reply:
If they are identifiers, should they be in the same namespace as the storage adapter?

Reply:
If they are in the same namespace, how can we track the difference between the file identifier and the file version identifier?